components/menu/src/auro-menu.js

```diff
@@ -307,222 +480,198 @@ export class AuroMenu extends LitElement {
   }
 
   /**
-   * Initializes all menu options in the DOM. This must be re-run every time the options are changed.
+   * Makes a selection based on the current index or clicked option.
    * @private
    */
-  initItems() {
-    this.items = Array.from(this.querySelectorAll('auro-menuoption, [auro-menuoption]'));
-    this.handleNoCheckmarkAttr();
-  }
+  makeSelection() {
+    if (!this.items) {
+      this.initItems();
+    }
 
-  /**
-   * Sets the index value of the selected item or first non-disabled menuoption.
-   * @private
-   */
-  getSelectedIndex() {
-    // find the first `selected` and not `disabled`, `hidden` or `static` option
-    const index = this.items.findIndex((option) => option.hasAttribute('selected') && this.optionInteractive(option));
+    const option = this.items[this.index];
+    if (option && !option.hasAttribute('disabled')) {
+      if (this.isOptionSelected(option)) {
+        // If the option is already selected, deselect it
+        this.handleDeselectState(option);
```

Comment:
This deselect should only happen when the menu is `multiselect` when more than one option is already selected.

e.g. clicking options is not a valid way to `reset` the menu to an initial state.